gatsby-node.js

```diff
@@ -84,6 +92,76 @@ const constructDbFilter = () => ({
     commit_hash: process.env.COMMIT_HASH || { $exists: false },
 });
 
+const createTagPageType = async (createPage, pageMetadata, stitchType) => {
+    const isAuthor = stitchType === 'author';
+    const pageType = STITCH_TYPE_TO_URL_PREFIX[stitchType];
+    const res = await stitchClient.callFunction('getValuesByKey', [
+        metadata,
+        stitchType,
+    ]);
+
+    const requests = [];
+
+    await res.forEach(async item => {
+        const requestKey = {};
+        requestKey[stitchType] = item._id;
+        requests.push(
+            stitchClient.callFunction('fetchDevhubMetadata', [
+                metadata,
+                requestKey,
+            ])
+        );
+    });
+
+    const pageData = await Promise.all(requests);
+
+    // Once requests finish, map the item with name (and optional image) to the response's return value
+    const itemsWithPageData = res.map((r, i) => ({
+        item: r,
+        pages: pageData[i],
+    }));
```

Comment:
Not sure if there is a cleaner way to avoid race conditions with network data other than to just wait for requests and map with initial objects.

Reply:
Not really. I think there are a couple request libraries that might make it easier, but probably not worth it since it's build side requests rather than the client